### PR TITLE
GCW-3252 New Scandit barcode scanner

### PR DESCRIPTION
**Pull request by patrixr**
### Ticket Link: 

https://jira.crossroads.org.hk/browse/GCW-3252
https://jira.crossroads.org.hk/browse/GCW-3209

### What does this PR do?

This PR substitutes the old barcode scanner with the [Scandit](scandit.com) scanner.

Differences:
- Only QR codes are readable now
- We can do continuous scanning on the Stocktake page
- We can integrate a preview within the page

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -224,7 +224,7 @@ jobs:
                 exit 1
             esac
             echo "Building for ENVIRONMENT: ${ENVIRONMENT}"
-            EMBER_CLI_CORDOVA=1 ENVIRONMENT=${ENVIRONMENT} yarn run ember build --environment=production
+            EMBER_CLI_CORDOVA=1 SCANDIT_LICENSE_KEY=${SCANDIT_LICENSE_KEY_ANDROID} ENVIRONMENT=${ENVIRONMENT} yarn run ember build --environment=production
```
Comment (abulsayyad123) [inline .circleci/config.yml:227]: is KEY accessible from azure ?

File: app/components/scan-barcode-button.js [hunk 2]
```diff
@@ -1,71 +1,55 @@
...
-      if (status.hasPermission) {
-        _this.scan(route);
-      } else {
-        //request permission here
-        permissions.requestPermission(
-          permissions.CAMERA,
-          permissionSuccess,
-          permissionError
-        );
+  async checkPermissionAndScan(route) {
```
Comment (abulsayyad123) [inline app/components/scan-barcode-button.js:11]: Clean !👌

File: .circleci/config.yml (hunk 1 above)
Comment (patrixr) [reply .circleci/config.yml:227]: I've added it to the circle ci ENV

File: app/utils/barcode-ui.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import Ember from "ember";
```
Comment (abulsayyad123) [inline app/utils/barcode-ui.js:1]: 👍

File: app/utils/cache.js [hunk 4]
```diff
@@ -4,50 +4,47 @@ import _ from "lodash";
...
-    return Ember.RSVP.resolve(func.apply(scope, args))
-      .then(_.noop)
-      .catch(e => {
-        _.pull(CACHE_IDS, cacheId);
-        return Ember.RSVP.reject(e);
-      });
+/**
+ * @description
+ *  <br> Will call func only if it hasn't been exectuted before
+ *  <br> Warning: This method does not return the value of the exectured method
```
Comment (abulsayyad123) [inline app/utils/cache.js:27]: misspelling is guess.

Comment (patrixr) [reply app/utils/cache.js:27]: 🙈

File: app/services/cordova.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ export default Ember.Service.extend({
   },
 
   appLoad() {
-    if (!config.cordova.enabled) {
+    if (!config.cordova.enabled || config.environment === "development") {
```
Comment (steveyken) [inline app/services/cordova.js:27]: still needed?

Comment (patrixr) [reply app/services/cordova.js:27]: No I can remove that, will do that now

**Pull request merged**